syntax = "proto3";

package hyperagent.manager;

import "common.proto";

service Manager {
    rpc RegisterAgent(RegisterAgentRequest) returns (RegisterAgentResponse);
    rpc SendTask(TaskRequest) returns (TaskResponse);
    rpc GetLogs(TaskRequest) returns (stream hyperagent.common.LogLine);
}

message RegisterAgentRequest {
    string agent_name = 1;
    repeated string capabilities = 2;
}

message RegisterAgentResponse {
    bool accepted = 1;
    string agent_id = 2;
}

message TaskRequest {
    string agent_id = 1;
    string task_type = 2;
    string payload = 3;
}

message TaskResponse {
    bool ok = 1;
    string task_id = 2;
}
